dspace-server-webapp/src/test/java/org/dspace/app/rest/ResourcePolicyRestRepositoryIT.java

```diff
@@ -2723,4 +2725,102 @@ public void resourcepolicySerchMethodWithPluralModelTest() throws Exception {
+                             .andExpect(jsonPath("$._embedded.eperson.id", is(newEPerson.getID().toString())))
+                             .andExpect(jsonPath("$._embedded.group", nullValue()));
+    }
+
+    @Test
+    public void patchReplaceGroupAdminTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        Group originGroup = GroupBuilder.createGroup(context)
+                                        .withName("origin Test Group")
+                                        .build();
+
+        Group newGroup = GroupBuilder.createGroup(context)
+                                     .withName("testGroupName")
+                                     .build();
+
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                                          .withName("Parent Community")
+                                          .build();
+
+        Collection col = CollectionBuilder.createCollection(context, parentCommunity)
+                                          .withName("Collection 1")
+                                          .build();
+
+        ResourcePolicy resourcePolicy = ResourcePolicyBuilder.createResourcePolicy(context)
+                                            .withAction(Constants.ADD)
+                                            .withDspaceObject(col)
+                                            .withGroup(originGroup)
+                                            .withDescription("My Description")
+                                            .withPolicyType(ResourcePolicy.TYPE_CUSTOM)
+                                            .build();
+
+        context.restoreAuthSystemState();
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+
+        // verify origin resourcepolicy
+        getClient(tokenAdmin).perform(get("/api/authz/resourcepolicies/" + resourcePolicy.getID()))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$._embedded.group.id", is(originGroup.getID().toString())))
+                             .andExpect(jsonPath("$._embedded.eperson", nullValue()));
+
+        // update group of the resourcePolicy
+        getClient(tokenAdmin).perform(post("/api/authz/resourcepolicies/" + resourcePolicy.getID() + "/group")
+                             .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                             .content("/api/eperson/groups/" + newGroup.getID()))
+                             .andExpect(status().isNoContent());
+
+        // verify that the resourcePolicy is related to new group
+        getClient(tokenAdmin).perform(get("/api/authz/resourcepolicies/" + resourcePolicy.getID()))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$._embedded.group.id", is(newGroup.getID().toString())))
+                             .andExpect(jsonPath("$._embedded.eperson", nullValue()));
+    }
+
```

Comment:
I woud add further scenario to be sure one and only one group or eperson can be associated to the policy, and to check that a policy related to group could not be changed and associated to a person and vice versa.